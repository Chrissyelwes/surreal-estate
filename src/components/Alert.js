import React from "react";
import PropTypes from "prop-types";

// eslint-disable-next-line react/prop-types
const Alert = ({ message, success }) => {
  return (
    <div className={`alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
