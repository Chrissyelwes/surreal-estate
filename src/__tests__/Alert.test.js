import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

test("displays an error message", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
  expect(asFragment()).toMatchSnapshot();
});

test("displays a success message", () => {
  const { getByText, asFragment } = render(
    <Alert message="Success!!!!" success />
  );

  expect(getByText(/Success/).textContent).toBe("Success!!!!");
  expect(asFragment()).toMatchSnapshot();
});

test("does not render an error or a success message if message props is empty", () => {
  const { asFragment } = render(<Alert message="" />);

  expect(asFragment()).toMatchSnapshot();
});
