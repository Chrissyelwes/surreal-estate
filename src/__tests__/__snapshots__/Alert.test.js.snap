// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays a success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!!!
  </div>
</DocumentFragment>
`;

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render an error or a success message if message props is empty 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  />
</DocumentFragment>
`;
